gov2/lambda/scenarios/scenario_get_started_functions.go

```diff
@@ -335,4 +284,51 @@ func (scenario GetStartedFunctionsScenario) Cleanup(ctx context.Context, role *i
 	}
 }
 
+// IScenarioHelper abstracts I/O and wait functions from a scenario so that they
+// can be mocked for unit testing.
+type IScenarioHelper interface {
+	Pause(secs int)
+	CreateDeploymentPackage(sourceFile string, destinationFile string) *bytes.Buffer
+}
+
+// ScenarioHelper lets the caller specify the path to Lambda handler functions.
+type ScenarioHelper struct {
+	HandlerPath string
+}
+
+// Pause waits for the specified number of seconds.
+func (helper *ScenarioHelper) Pause(secs int) {
+	time.Sleep(time.Duration(secs) * time.Second)
+}
+
+// CreateDeploymentPackage creates an AWS Lambda deployment package from a source file. The
+// deployment package is stored in .zip format in a bytes.Buffer. The buffer can be
+// used to pass a []byte to Lambda when creating the function.
+// The specified destinationFile is the name to give the file when it's deployed to Lambda.
+func (helper *ScenarioHelper) CreateDeploymentPackage(sourceFile string, destinationFile string) *bytes.Buffer {
+	var err error
+	buffer := &bytes.Buffer{}
+	writer := zip.NewWriter(buffer)
+	zFile, err := writer.Create(destinationFile)
+	if err != nil {
+		log.Panicf("Couldn't create destination archive %v. Here's why: %v\n", destinationFile, err)
+	}
+	sourceBody, err := os.ReadFile(fmt.Sprintf("%v/%v", helper.HandlerPath, sourceFile))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

User input controlling server file access enables path traversal, allowing access to unintended files. Sanitize and validate file paths before usage. Recommend allowlisting permitted paths and filenames. Never concatenate or join user input into filesystem paths. Input validation prevents malicious actors from accessing sensitive files.